sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java

```diff
@@ -78,10 +96,24 @@ public void stopBroker() throws Exception {
 
   @Test
   @Category(NeedsRunner.class)
+  public void testAuthenticationRequired() {
+    expectedException.expect(Exception.class);
+    expectedException.expectMessage("JMSSecurityException");
```

Comment:
Optional:

how good is the error message returned here? Does it say "you must log in"? or just it just does "JMSSecurityException"?

For user experience, we should try to make sure it's a good error. And if so, maybe we should test some key phrase like "log in" if there is a good one.